### PR TITLE
Fix stackplots static style

### LINKED ISSUE #4912: Navigating from condition widget back to display layout can cause background to become styled with Localstorage
#### Summary
Navigating between a condition widget and a layout can cause the background of the entire layout to change color, _even though no background styling has been defined for the layout._

Note: this only appears to affect localStorage

#### Steps to Reproduce
1. Import the attached JSON file
2. Switch to real-time mode in the conductor
3. Navigate to My Items > My Items > Components > Demo Conditions > Battery Widget
4. Go into edit mode
5. In the inspector, click "styles"
6. Cancel out of edit mode
7. Navigate to My Items > My Items > Components > Demo Tabs View > Spacecraft Plots

#### Environment

* Open MCT Version: 74b83903c907680ea183d4ec331fb649d9a9fcc4
* Deployment Type: local
* Browser: Chrome 98.0.4758.109

#### Impact Check List

- [X] Data loss or misrepresented data? - Yes can make it appear as though a condition has been triggered when it hasn't
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [X] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?

#### Additional Information
[My Items - condition bug.json.zip](https://github.com/nasa/openmct/files/8181434/My.Items.-.condition.bug.json.zip)

Screen Shot 2022-03-03 at 1 58 58 PM

### PR DESCRIPTION
**Pull request by shefalijoshi**
Closes #4912 #4864 

### Describe your changes:
The StyleRuleManager was subscribing to condition set updates even when it was destroyed (via a promise). Now we check to make sure that the StyleRuleManager is not destroyed when we handle the promise.
Cancelling the properties dialog was not handled. Added a handler for this.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate unit tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Commit messages meet standards?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [x] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by davetsay** [comment]
sorry @shefalijoshi , i think i let this get stale and now its out of sync with master

**Comment by davetsay** [comment]
@nikhilmandlik , can you take a look at this one?

File: src/plugins/condition/StyleRuleManager.js [hunk 1]
```diff
@@ -78,11 +78,13 @@ export default class StyleRuleManager extends EventEmitter {
         this.openmct.objects.get(this.conditionSetIdentifier).then((conditionSetDomainObject) => {
             this.openmct.telemetry.request(conditionSetDomainObject)
                 .then(output => {
-                    if (output && output.length) {
+                    if (output && output.length && (this.conditionSetIdentifier && this.openmct.objects.areIdsEqual(conditionSetDomainObject.identifier, this.conditionSetIdentifier))) {
```
Comment (nikhilmandlik) [inline src/plugins/condition/StyleRuleManager.js:81]: I think can add following before , `this.openmct.objects.get( .......`
```
if (!this.conditionSetIdentifier) {
      return;
}
```

File: src/plugins/formActions/EditPropertiesAction.js [hunk 2]
```diff
@@ -76,6 +76,13 @@ export default class EditPropertiesAction extends PropertiesAction {
         }
     }
 
+    /**
+     * @private
+     */
+    _onCancel() {
+        //noop
```
Comment (nikhilmandlik) [inline src/plugins/formActions/EditPropertiesAction.js:83]: ahh thanks, 
there are more like this tracked here, https://github.com/nasa/openmct/issues/4864

**Review by nikhilmandlik — CHANGES_REQUESTED** [review]
very small change.

File: src/plugins/condition/StyleRuleManager.js (hunk 1 above)
Comment (shefalijoshi) [reply src/plugins/condition/StyleRuleManager.js:81]: Ah unfortunatley, this is happening if this.conditionSetIdentifier gets unset AFTER the objects.get happens. That's why I added this check. :/

**Review by nikhilmandlik — APPROVED** [review]
LGTM

**Comment by nikhilmandlik** [comment]
Needs testing fix

**Pull request merged**